### PR TITLE
fix(editor): Use spaces for tabs when indentWithTabs is false

### LINKED ISSUE #1238: tab behavior should match autoindent
out

At least while editing Python code, pressing tab should result in 4 spaces. Regardless of language, autoindent should insert the type of whitespace pressing tab would give the user so they can mix automatically indented lines with lines manually indented with tab.

Related to https://github.com/nteract/nteract/issues/1137, but there may be more to that issue.

### PR DESCRIPTION
**Pull request by captainsafia**
Closes #1238.

nteract-indent

### CODE REVIEW

**Comment by rgbkrk** [comment]
Hurrah!

However, it looks like you'll need to rebase against origin/master.

**Comment by captainsafia** [comment]
Bleh. I dunno what's wrong with my rebases. I did a rebase against master and origin/master but none seem to wipe out the commits.

**Comment by captainsafia** [comment]
OK I FIGURED OUT HOW TO DO IT AND MY GIT SKILLS HAVE LEVELED UP WOW.

Give me two seconds to clean this up then it should be good for review and merge.

**Comment by rgbkrk** [comment]
level up

**Comment by captainsafia** [comment]
OK THAT ONLY WORKED FOR SOME COMMITS LET ME TRY A DIFFERENT THING FOR THE OTHERS.

Wow.

This is a saga.

Safia and the Very Messy Rebase.

**Comment by captainsafia** [comment]
IT IS DONE MERGE IT!

File: src/notebook/components/cell/editor/index.js [hunk 1]
```diff
@@ -272,6 +272,15 @@ export default class Editor extends React.Component {
       .subscribe(callback);
   }
 
+  tabToSpaces(editor: Object): void {
+    if (editor.somethingSelected()) {
+      editor.indentSelection('add');
+    } else {
+      editor.replaceSelection(editor.getOption('indentWithTabs') ? '\t' :
+        Array(this.props.tabSize + 1).join(' '), 'end', '+input');
+    }
```
Comment (rgbkrk) [inline src/notebook/components/cell/editor/index.js:281]: I'm guessing this comes from https://github.com/codemirror/CodeMirror/issues/988#issuecomment-14921785

That thread had a lot of discourse...

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
Little drop in coverage, @captainsafia promised to do tests later. :smile:

**Pull request merged**